src/plugins/data/public/index_patterns/fields/field.test.ts

```diff
@@ -0,0 +1,223 @@
+    lang: 'lang',
+    count: 1,
+    esTypes: ['type'],
+    aggregatable: true,
+    filterable: true,
+    searchable: true,
+    sortable: true,
+    readFromDocValues: false,
+    visualizable: true,
+    scripted: true,
+    subType: { multi: { parent: 'parent' }, nested: { path: 'path' } },
+    displayName: 'displayName',
+    indexPattern: ({
+      fieldFormatMap: { name: {}, _source: {}, _score: {}, _id: {} },
+    } as unknown) as IndexPattern,
+    format: { name: 'formatName' },
+    $$spec: {},
+    conflictDescriptions: { a: ['b', 'c'], d: ['e'] },
+  } as Field;
+
+  it('the correct properties are writable', function () {
+    const field = getField();
+
+    expect(field.count).toEqual(1);
+    field.count = 2;
+    expect(field.count).toEqual(2);
+
+    expect(field.script).toEqual(fieldValues.script);
+    field.script = '1';
+    expect(field.script).toEqual('1');
+
+    expect(field.lang).toEqual(fieldValues.lang);
+    field.lang = 'painless';
+    expect(field.lang).toEqual('painless');
+
+    expect(field.conflictDescriptions).toEqual(fieldValues.conflictDescriptions);
+    field.conflictDescriptions = {};
+    expect(field.conflictDescriptions).toEqual({});
+  });
+
+  it('the correct properties are not writable', function () {
+    const field = getField();
+
+    expect(field.name).toEqual(fieldValues.name);
+    field.name = 'newName';
+    expect(field.name).toEqual(fieldValues.name);
+
+    expect(field.type).toEqual(fieldValues.type);
+    field.type = 'newType';
+    expect(field.type).toEqual(fieldValues.type);
+
+    expect(field.esTypes).toEqual(fieldValues.esTypes);
+    field.esTypes = ['newType'];
+    expect(field.esTypes).toEqual(fieldValues.esTypes);
+
+    expect(field.scripted).toEqual(fieldValues.scripted);
+    field.scripted = false;
+    expect(field.scripted).toEqual(fieldValues.scripted);
+
+    expect(field.searchable).toEqual(fieldValues.searchable);
+    field.searchable = false;
+    expect(field.searchable).toEqual(fieldValues.searchable);
+
+    expect(field.aggregatable).toEqual(fieldValues.aggregatable);
+    field.aggregatable = false;
+    expect(field.aggregatable).toEqual(fieldValues.aggregatable);
+
+    expect(field.readFromDocValues).toEqual(fieldValues.readFromDocValues);
+    field.readFromDocValues = true;
+    expect(field.readFromDocValues).toEqual(fieldValues.readFromDocValues);
+
+    expect(field.subType).toEqual(fieldValues.subType);
+    field.subType = {};
+    expect(field.subType).toEqual(fieldValues.subType);
+
+    // not writable, not serialized
+    expect(() => {
+      field.indexPattern = {} as IndexPattern;
+    }).toThrow();
+
+    // computed fields
+    expect(() => {
+      field.format = { name: 'newFormatName' };
+    }).toThrow();
+
+    expect(() => {
+      field.sortable = false;
+    }).toThrow();
+
+    expect(() => {
+      field.filterable = false;
+    }).toThrow();
+
+    expect(() => {
+      field.visualizable = false;
+    }).toThrow();
```

Comment:
nit: Is there a specific error that's expected here? Could use `toThrowErrorMatchingInlineSnapshot()`

Reply:
Good suggestion but I think I'll leave this be for now as it will soon be enforced by typescript.